scala/scala-impl/src/org/jetbrains/plugins/hydra/compiler/ScalaHydraCompilerConfigurationPanel.scala

```diff
@@ -103,4 +138,9 @@ class ScalaHydraCompilerConfigurationPanel(project: Project, settings: HydraComp
   private def downloadVersion(scalaVersions: Seq[String], hydraVersion: String): (String => Unit) => Unit =
     (listener: (String) => Unit) => scalaVersions.foreach(version => HydraArtifactsCache.downloadIfNotPresent(version, hydraVersion, listener))
 
+  private def setDefaultHydraVersion: Unit = {
+    val hydraVersions = HydraVersions.downloadHydraVersions
```

Comment:
I'm on a train right now with a slow/bad connection and I'm experiencing some really bad sluggishness when I edit the Hydra settings panel. It's so bad that I basically have to kill and restart idea.

Reply:
I experienced this behavior too. I also had a slow connection so because the Hydra versions are downloaded every time you change your username or the repository for example. Maybe this is not the best solution. And also the Hydra settings panel opens very slow if you have a bad connection because of the version downloading.

Reply:
Indeed. It seems like having a plain text field is best, but if you have something else in mind let us know.

Reply:
Maybe a separate dialog after clicking the download button. In this dialog you are asked to choose the Hydra version. Then the version download only happens when you click download.